### ISSUE TITLE
[9.0] Error when removing all contacts from the merge contacts wizard

### ISSUE THREAD

**Issue by krueger80** [odoo/odoo#11343]
#### Impacted versions:
[9.0]

#### Steps to reproduce:
1) Create a new database with demo data
2) Install crm application
3) Go in Contacts app
4) Select Agrolait
5) Go in Action->Merge Selected Contacts
6) Click on the bin to remove the selected contact
7) Click on the Destination Contact field

#### Current behavior:
JS error:
```
Uncaught Error: No conversion for undefined
```
```
http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:3103
Traceback:
Error: No conversion for undefined
    at wrap (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:3103:44)
    at Object.py.type.__getitem__ (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:3108:293)
    at Object.py.PY_getItem (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2889:11)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2972:74)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2967:123)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2967:204)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2968:43)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2971:111)
    at Object.py.evaluate (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2973:99)
    at Object.py.eval (http://localhost:8069/web/content/381-34903e6/web.assets_backend.js:2978:284)
```

#### Expected behavior:
I should not see nothing in the list instead of an error.

#### References:
#11295

**Comment by krueger80** [comment]
So you simply leaving this bad code un Odoo? I simply want you to fix it so I can have a reference on how to code right. If the solution you proposed works, fix it, but it don't.

Having a domain based on a x2many is important in what I'm developping and you don't seem to take it seriously.

**Comment by beledouxdenis** [comment]
I do not follow what you say 

1aeb747c6e06f11a005331713e5858438cdf8a60 isn't solving the issue at your end ?

**Comment by krueger80** [comment]
No as I say in other cases the onchange doesn't run when getting in an existing record. So the drop down will not be filtered until a real change is made on the field where the onchange is placed. Not good.

In this particular case, this is the only place in your code where you used the technique I want to use to avoid the previous issue.

**Comment by krueger80** [comment]
Oups let me check the change first! I didn't see that!

**Comment by krueger80** [comment]
Thanks!!! You made my day! I'll try it at work.

**Comment by krueger80** [comment]
Sorry for the confusion. On the mobile version of github I didn't see clearly that you made a commit to fix that... I though you simply closed it. So it worked. By the way, it fixes my other issue #11295. Thanks.
